docs/native-modules-android.md

```diff
@@ -650,17 +998,112 @@ public class ImagePickerModule extends ReactContextBaseJavaModule {
 }
 ```
 
+</TabItem>
+<TabItem value="kotlin">
+
+```kotlin
+class ImagePickerModule(reactContext: ReactApplicationContext) :
+    ReactContextBaseJavaModule(reactContext) {
+
+    private var pickerPromise: Promise? = null
+
+    private val activityEventListener =
+        object : BaseActivityEventListener() {
+            override fun onActivityResult(
+                activity: Activity?,
+                requestCode: Int,
+                resultCode: Int,
+                intent: Intent?
+            ) {
+                if (requestCode == IMAGE_PICKER_REQUEST) {
+                    pickerPromise?.let { promise ->
+                        when (resultCode) {
+                            Activity.RESULT_CANCELED ->
+                                promise.reject(E_PICKER_CANCELLED, "Image picker was cancelled")
+                            Activity.RESULT_OK -> {
+                                val uri = intent?.data
+
+                                uri?.let { promise.resolve(uri.toString()) }
+                                    ?: promise.reject(E_NO_IMAGE_DATA_FOUND, "No image data found")
+                            }
+                        }
+
+                        pickerPromise = null
+                    }
+                }
+            }
+        }
+
+    init {
+        reactContext.addActivityEventListener(activityEventListener)
+    }
+
+    override fun getName() = "ImagePickerModule"
+
+    @ReactMethod
+    fun pickImage(promise: Promise) {
+        val activity = currentActivity
+
+        if (activity == null) {
+            promise.reject(E_ACTIVITY_DOES_NOT_EXIST, "Activity doesn't exist")
+            return
+        }
+
+        pickerPromise = promise
+
+        try {
+            val galleryIntent = Intent(Intent.ACTION_PICK).apply { type = "image\/*" }
+
+            val chooserIntent = Intent.createChooser(galleryIntent, "Pick an image")
+
+            activity.startActivityForResult(chooserIntent, IMAGE_PICKER_REQUEST)
+        } catch (t: Throwable) {
+            pickerPromise?.reject(E_FAILED_TO_SHOW_PICKER, t)
+            pickerPromise = null
+        }
+    }
+}
+
+    companion object {
+        const val IMAGE_PICKER_REQUEST = 1
+        const val E_ACTIVITY_DOES_NOT_EXIST = "E_ACTIVITY_DOES_NOT_EXIST"
+        const val E_PICKER_CANCELLED = "E_PICKER_CANCELLED"
+        const val E_FAILED_TO_SHOW_PICKER = "E_FAILED_TO_SHOW_PICKER"
+        const val E_NO_IMAGE_DATA_FOUND = "E_NO_IMAGE_DATA_FOUND"
+    }
+}
```

Comment:
Hmm, it looks like there is an additional closing bracket (`}`), which leads to the error, when trying to format the code in `ktfmt` tool.